### PR TITLE
Use custom error action regardless of response format

### PR DESCRIPTION
**Pull request by erikverheij**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | 

`Yii\web\ErrorHandler` has the ability to set a custom action for rendering errors. Now it uses that action regardless of the response format. Previously the custom action was not triggered when the response format was something other than `FORMAT_HTML` which is unexpected behavior.

### CODE REVIEW

**Review by SilverFire — APPROVED** [review]
Except minor changes in CHANGELOG

**Comment by cebe** [comment]
this changes existing behavior, so it is a BC break.

**Comment by erikverheij** [comment]
Thank you for reviewing this PR. Is there any action from my side required?

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ Yii Framework 2 Change Log
 
 2.0.11 under development
 ------------------------
-
+- Bug: Use custom error action when provided, regardless of the response format.
```
Comment (samdark) [inline framework/CHANGELOG.md]: Should go to 2.1 changelog.

**Comment by samdark** [comment]
I am OK changing that in 2.1. If user set error explicitly it is logical to do exactly what was configured.

**Comment by SilverFire** [comment]
Btw it will be good to have unit-tests for ErrorHandler.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**